src/main/java/de/epiceric/shopchest/listeners/ShopUpdateListener.java

```diff
@@ -23,48 +14,6 @@ public ShopUpdateListener(ShopChest plugin) {
         this.plugin = plugin;
     }
 
-    @EventHandler(priority = EventPriority.MONITOR, ignoreCancelled = true)
-    public void onPlayerTeleport(final PlayerTeleportEvent e) {
```

Comment:
Are you sure this method can be removed? If you teleport to a location far away, the packets to spawn the armor stands and items are sent too early when the client has not loaded the chunk yet, so the packets are ignored and no armor stand or item will be spawned.